modules/datahandling/filerequests.go

```diff
@@ -309,9 +318,12 @@ func (f fileChangeRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
 			"SenderID":  f.SenderID,
 			"ProjectID": fileMeta.ProjectID,
 		})
+		ack()
 		return []dhClosure{toSenderClosure{msg: messages.NewEmptyResponse(messages.StatusUnauthorized, f.Tag)}}, nil
 	}
 
+	ack() // ack just before appending so if it fails in the middle it doesn't re-enqueue it
```

Comment:
yeah you're right that it does. hmm, i guess I'm more concerned about it getting stuck than I am about it failing the cas check